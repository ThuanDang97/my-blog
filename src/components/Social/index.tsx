import { SvgIcon, useTheme } from '@mui/material'
import Link from 'next/link'
import { memo, ReactNode } from 'react'
import {
  RiFacebookFill,
  RiGithubFill,
  RiInstagramLine,
  RiLinkedinFill,
  RiMailFill,
  RiTwitterFill,
} from 'react-icons/ri'

const platformIcons: Record<string, ReactNode> = {
  facebook: <RiFacebookFill />,
  instagram: <RiInstagramLine />,
  twitter: <RiTwitterFill />,
  linkedin: <RiLinkedinFill />,
  gmail: <RiMailFill />,
  github: <RiGithubFill />,
}

export type TPlatform = {
  slug: string
  link: string
}
const Social = ({ slug, link }: TPlatform) => {
  const theme = useTheme()

  return (
    <Link href={link} passHref>
      <SvgIcon
        sx={{
          color: theme.palette.common.black,
          width: 18,
          height: 18,
        }}
      >
        {platformIcons[slug]}
      </SvgIcon>
    </Link>
  )
}

export default memo(Social)
