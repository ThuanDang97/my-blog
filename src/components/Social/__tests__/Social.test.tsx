import { render } from '@utils/testUtils'
import Social, { TPlatform } from '..'

describe('Social Component', () => {
  const mockProps: TPlatform = {
    slug: 'facebook',
    link: '#',
  }

  it('Should show match Social DOM Snapshot', () => {
    const { container } = render(<Social {...mockProps} />)

    expect(container).toMatchSnapshot()
  })
})
