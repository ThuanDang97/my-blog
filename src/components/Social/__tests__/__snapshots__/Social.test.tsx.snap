// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social Component Should show match Social DOM Snapshot 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-m7rsiw-MuiSvgIcon-root"
    focusable="false"
    href="/#"
    viewBox="0 0 24 24"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M14 13.5h2.5l1-4H14v-2c0-1.03 0-2 2-2h1.5V2.14c-.326-.043-1.557-.14-2.857-.14C11.928 2 10 3.657 10 6.7v2.8H7v4h3V22h4v-8.5z"
        />
      </g>
    </svg>
  </svg>
</div>
`;
